pages/static/survey.js

```diff
@@ -0,0 +1,255 @@
+      const el = e.target;
+      let card = el;
+
+      while (card && !card.classList.contains('active')) {
+        card = card.parentNode;
+      }
+
+      const index = card.parentNode.children.indexOf(card);
+      const question = this.questions[index];
+
+      if (question.type === 'likert' || question.type === 'rating') {
+        question.answer = el.nextSibling.textContent;
+      } else if (question.type === 'open') {
+        question.answer = card.querySelector('textarea').value;
+      }
+
+      this.nextSlide();
+    });
+  });
+
+  dismiss.addEventListener('click', this.submit);
+
+  setTimeout(this.nextSlide, 500);
+};
+
+Fez.prototype.nextSlide = async function () {
+  const cards = form.querySelectorAll('.slide');
+  const currentCard = form.querySelector('.active');
+  let currentIndex = cards.indexOf(currentCard);
+  const nextIndex = (currentIndex += 1);
+
+  if (nextIndex < cards.length) {
+    if (currentCard) {
+      currentCard.classList.remove('active');
+    }
+
+    cards[nextIndex].classList.add('active');
+
+    if (nextIndex === cards.length - 1) {
+      dismiss.classList.add('hide');
+      this.submit();
+    }
+  }
+
+  const {height} = await cards[nextIndex].getBoundingClientRectAsync();
+
+  form.style.height = `${height}px`;
+};
+
+Fez.prototype.submit = async function (dismissed) {
+  $('#fez').style.height = 0;
+
+  try {
+    fetch(`${self.location.origin}/fez-survey-response`, {
+      method: 'POST',
+      cache: 'no-cache',
+      headers: {
+        'Content-Type': 'application/json',
+      },
+      body: JSON.stringify({
+        'survey': this.name,
+        'questions': this.questions,
+        'url': this.url,
+        'shownAt': `${this.shownAt}`,
+        dismissed,
+      }),
+    });
+  } catch (e) {
+    // we don't actually care if there are errors at this point
+    // they will show up in the server logs, and we don't want
+    // to show any error to the end user.
+  }
+
+  // store the fact we replied to the survey for 4 weeks
+  localStorage.setItem(
+    `survey_${this.name}`,
+    Date.now(this.shownAt) + 1000 * 60 * 60 * 24 * 28
+  );
+
+  this.remove(dismissed);
+};
+
+Fez.prototype.remove = function (dismissed) {
+  form.style.height = 0;
+  setTimeout(
+    () => {
+      form.parentNode.innerHTML = '';
+    },
+    dismissed ? 0 : 2500
+  );
+};
+
+Fez.prototype.build = function () {
+  const slides = $('.slides');
+  const thanks = $('.thanks').parentNode;
+
+  this.questions.forEach((q, i) => {
+    slides.insertBefore(new FezSlide(q, i), thanks);
+  });
+
+  this._attachListeners();
+};
+
+function FezSlide(data, index) {
+  const wrap = c('div');
+  const question = c('p');
+
+  wrap.classList.add('slide');
+  question.classList.add('surveyQuestion');
+
+  wrap.appendChild(question);
+  question.textContent = data.text;
+
+  if (data.type === 'rating' || data.type === 'likert') {
+    const ul = c('ul');
+    const values = data.values || [
+      'Strongly Agree',
```

Comment:
How can these be localised?

Reply:
yes, I am not sure what the best way to store those localized strings would be, but they will be

Reply:
You could go the same route we went with Pixi: put all strings in a `amp-state` element and then get them from there.

https://github.com/ampproject/amp.dev/blob/18f70f87722f8443e4587379ce98405946b97868/pages/content/amp-dev/page-experience.html#L9-L22